buildSrc/build.gradle

```diff
@@ -16,23 +16,41 @@
  * limitations under the License.
  */
 
-// Define the set of repositories and dependencies required to
-// fetch and enable plugins.
-buildscript { dependencies { classpath "com.diffplug.spotless:spotless-plugin-gradle:3.17.0" } }
-
 // Plugins for configuring _this build_ of the module
-plugins { id 'groovy' }
-
-apply plugin: "com.diffplug.gradle.spotless"
+plugins {
+  id 'java-gradle-plugin'
+  id 'groovy'
+  id "com.diffplug.gradle.spotless" version "3.17.0"
+}
 
-// Dependencies on other plugins used when this plugin
-// is invoked
-repositories { jcenter() }
+// Define the set of repositories required to fetch and enable plugins.
+repositories {
+  jcenter()
+  maven { url "https://plugins.gradle.org/m2/" }
+  maven { url "http://repo.spring.io/plugins-release" }
+}
 
+// Dependencies on other plugins used when this plugin is invoked
 dependencies {
   compile gradleApi()
   compile localGroovy()
   compile 'com.github.jengelman.gradle.plugins:shadow:4.0.3'
+
+  runtime "net.ltgt.gradle:gradle-apt-plugin:0.20"                                                  // Enable a Java annotation processor
```

Comment:
Wow, this is much better to have these dependencies adjacent to the code that needs them. Great!